src/java/org/apache/cassandra/metrics/ClientMetrics.java

```diff
@@ -118,6 +193,31 @@ public long getCount()
         authSuccess = registerMeter("AuthSuccess");
         authFailure = registerMeter("AuthFailure");
 
+        // For each of SSL, non-SSL register a gauge:
+        encryptedConnectedNativeClients = registerGauge(new DefaultNameFactory("Client", "Encrypted"), CONNECTED_NATIVE_CLIENTS, () -> countConnectedClients((ServerConnection::isSSL)));
+        unencryptedConnectedNativeClients = registerGauge(new DefaultNameFactory("Client", "Unencrypted"), CONNECTED_NATIVE_CLIENTS, () -> countConnectedClients(((ServerConnection connection) -> !connection.isSSL())));
+
+        // for each supported authentication mode, register a meter for success and failures.
+        final IAuthenticator authenticator = DatabaseDescriptor.getAuthenticator();
+        for (String mode : authenticator.getSupportedAuthenticationModes())
+        {
+            String capitalizedMode = StringUtils.capitalize(mode);
```

Comment:
we can't we save the mode capitalized already? If `mode` was an enum, you would just used its name.

Reply:
:+1: agreed, better to leave it up to the implementors